### PR TITLE
Provider name in WalletComponent

### PR DESCRIPTION
**Pull request by Velenir**
Adds provider name or wallet name (for WalletConnect)

localhost_8080_trade_DAI-USDC_sell=0 price=0 from= expires= (6)
localhost_8080_trade_DAI-USDC_sell=0 price=0 from= expires= (5)
localhost_8080_trade_DAI-USDC_sell=0 price=0 from= expires= (4)
localhost_8080_trade_DAI-USDC_sell=0 price=0 from= expires= (3)
localhost_8080_trade_DAI-USDC_sell=0 price=0 from= expires= (2)

### CODE REVIEW

**Comment by W3stside** [comment]
ooooh nice, will review soon

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr1270--dexreact.review.gnosisdev.com

File: src/api/wallet/WalletApi.ts [hunk 1]
```diff
@@ -515,6 +521,19 @@ export class WalletApiImpl implements WalletApi {
 
   /* ****************      Private Functions      **************** */
 
+  private _setProviderInfo(): void {
+    // this can get expensive depending on the number and complexity of checks in getProviderInfo
+    // so retrigger only on connect/disconnect
+    this._providerInfo = getProviderInfo(this._provider)
+
+    if (this._providerInfo && isWalletConnectProvider(this._provider) && this._provider.wc.peerMeta) {
```
Comment (W3stside) [inline src/api/wallet/WalletApi.ts:529]: what is the `peerMeta` for?

Comment (Velenir) [reply src/api/wallet/WalletApi.ts:529]: Probably lots of stuff. It's where WC stores [wallet info](https://github.com/WalletConnect/walletconnect-monorepo/blob/efd31442ca7251ccdef3ca4d832a8f01276bf74d/packages/helpers/types/index.d.ts#L183)
There's also `clientMeta` which seems to be the same

**Comment by Velenir** [comment]
Added displaying `peerMeta.icons[0]` when available
Will add icons for known providers next

**Comment by Velenir** [comment]
cbW
localhost_8080_trade_DAI-USDC_sell=0 price=0 from= expires= (9)
localhost_8080_trade_DAI-USDC_sell=0 price=0 from= expires= (8)
localhost_8080_trade_DAI-USDC_sell=0 price=0 from= expires= (7)
localhost_8080_trade_DAI-USDC_sell=0 price=0 from= expires= (6)
localhost_8080_trade_DAI-USDC_sell=0 price=0 from= expires= (3)
localhost_8080_trade_DAI-USDC_sell=0 price=0 from= expires= (2)
localhost_8080_trade_DAI-USDC_sell=0 price=0 from= expires= (1)


localhost_8080_trade_DAI-USDC_sell=0 price=0 from= expires= (4)
Coinomi is small, because they offer that icon in `peerMeta.icons[0]` https://www.coinomi.com/svg/coinomi-logo-color.svg

@anxolin can you check for rainbowWallet

**Comment by anxolin** [comment]
image
@Velenir nice job!

**Pull request merged**